ArduCopter/control_guided.cpp

```diff
@@ -575,6 +579,18 @@ void Copter::guided_posvel_control_run()
         pos_control->update_xy_controller(AC_PosControl::XY_MODE_POS_AND_VEL_FF, ekfNavVelGainScaler, false);
     }
 
+    float temp_surface_tracking_climbrate=0.0;
+    // run altitude controller
+    if (sonar_enabled && (sonar_alt_health >= SONAR_ALT_HEALTH_MAX)) {
+        // if sonar is ok, use surface tracking
+        //posvel_vel_target_cms.z = get_surface_tracking_climb_rate(posvel_vel_target_cms.z, pos_control.get_alt_target(), G_Dt);
+        temp_surface_tracking_climbrate = get_surface_tracking_climb_rate((int)posvel_vel_target_cms.z, posvel_pos_target_cm.z, G_Dt);
```

Comment:
Why not declare it as const float in this line and save the initialization with zero above ?